### ISSUE TITLE
11.7 Nutzer-Einstellungen - Beispiele

### ISSUE THREAD

**Issue by johannesFischer84** [BIK-BITV/BIK-Web-Test#24]
Ich würde es gut finden, wenn man für Firefox und Chrome auflisten könnte, wo man die Einstellungen für Farben und Schriften findet. Für die weiteren Optionen aus 11.7 (Einheiten, Kontrast, Fokus-Cursor)gibt es in den Browsern keine Einstellmöglichkeiten, wie es mir scheint.

Ein Entwurf für Farben und Schriften:
## Farben
Firefox-Einstellungen: Register Allgemein -> Sprache und Erscheinungsbild -> Schriftarten & Farben -> Farben. Im Auswahlfeld "Oben ausgewählte Farben anstatt der Farben der Seite verwenden" die Option "Immer" wählen.
In Chrome gibt es keine Farb-Einstellungen.
## Schriften
Firefox-Einstellungen: Register Allgemein -> Sprache und Erscheinungsbild -> Schriftarten & Farben -> Erweitert. Auswahlfeld "Seiten das Verwenden von eigenen statt der oben gewählten Schriftarten erlauben" deaktivieren. Für die Schriftgröße muss der Wert für die Mindestschriftgröße verändert werden. Sonst ändert sich die Größe nur, falls die Seite keine eigene Schriftgröße definiert. In den Feldern Serif / Sans Serif andere Schriftart wählen, Ok drücken.

Chrome-Einstellungen: Bereich Darstellung -> Schriftart anpassen. Für die Schriftgröße wieder die Mindestschriftgröße ändern.

Bei der Schriftart in Chrome ändert sich bei mir nur etwas, wenn die Seite im CSS keine eigene Schriftart definiert. Das scheint mir ein Bug in Chrome zu sein, oder ich habe die Auswahl übersehen, wo man den Browser zwingen kann, die definierte Schrift statt der Webseiten-Schrifteinstellung zu nutzen. Hat hier jemand etwas anderes gefunden?

**Comment by Martina-XX** [comment]
Hallo,
sorry fürs Hochholen von diesem etwas älteren Issue, aber ich denke, meine Frage passt hier gut hin.

Habt ihr spezielle Farbeinstellungen, mit denen ihr 11.7 testet? 
Je nachdem, welche Kombination man unter den Farbeinstellungen auswählt, könnte es ja zu einem Fail oder Pass führen. Es würde dann von den Einstellungen des Testers abhängen, ob dieser Teil des Prüfschritts bestanden ist oder nicht. 
Hat jemand da zufällig eine "optimale" Farbempfehlung oder eine spezielle Farbauswahl - und wenn ja, wieso gerade diese Farben? :) 

BTW: Im Prüfschritt steht, dass der Firefox die Farben für ein `button`-Element nicht übernimmt. Mittlerweile ändert Firefox die Farbe des Buttons - allerdings eigenständig. Ich glaube, Firefox entscheidet anhand der Hintergrundfarbe, welche Farbe er den Buttons zuweist (aber gefährliches, nicht geprüftes Halbwissen).

**Comment by JSchwehn** [comment]
Hi @Martina-XX Ich verstehe den Prúfschritt so, dass der Benutzende komplett frei einstellen kann, welche Farben und Font verwendet werden sollen. Wenn es hellblau auf dunkelblau ist, dann ist das so. Ich überprüfe nicht ob dann die z.B. die Kontraste passen. 

Für meine Tests habe ich versucht ein maximum an Kontrast zu erreichen, so das ich schnell sehen kann ob meine Einstellungen übernommen wurden und  auch überall Comic-Sans auftaucht :) 

Korregiert mich, wenn ich den Prüfschritt falsch verstanden habe.

**Comment by Martina-XX** [comment]
Ja, so verstehen wir den Prüfschritt auch - es sind alle möglichen Einstellungskombinationen des Nutzers möglich.

Kurz zum Hintergrund der Frage: 
Auf Testerebene wollen wir einen Wert festlegen, damit wir (Tester) dieselben Farben verwenden. Wir würden gern eine reale Kombination wählen, sind bei der Farbwahl aber ratlos.

@JSchwehn, wenn ich dich richtig verstehe, prüfst du "nur", ob die Farben übernommen werden. Du prüfst nicht, ob Inhalte dann noch wahrnehmbar sind? Oder doch? 
Das umgeht zumindest das Problem, wenn die Farbwahl nicht zueinander passt. 

Nutzt ihr ggf. auch verschiedene Farben bei unterschiedlichen Webseiten? Oder habt ihr immer nur eine Farbkombination?

(BTW: Comic Sans for the win! 😄 Mit der teste ich auch immer.)

**Comment by JSchwehn** [comment]
@Martina-XX Ja, das verstehst Du richtig. Ich teste hier nur ob die Einstellungen übernommen werden. Aus meiner Sicht ist der Jenige der die Einstellungen überschreiben will auch dafür verantwortlich das der Inhalt auch für ihn lesbar ist.

Bei der Farb- und Font-Auswahl haben wir uns entschieden die Priorität auf "schnell erkennen das was nicht übernommen wird" zu setzen. 

Ich habe ein extra Benuzer-Profil im Firefox angelegt mit extremen Farben und Fonts das ich auf allen Seiten verwende. Aber wir testen unsere eigenen Seiten und da ist die Variablität nicht sehr hoch.

Hier mal ein kleiner Screenshot auf dem der Mangel an ComicSans in 32px sofort zu sehen ist und ich kann garantieren die Farbkombination haben wir nirgends. 
Mein Vorschlag zur Farbauswahl wäre: kräftige Farben, die nicht auf der regulären Webseite vorkommen, wählen. Einen grossen Kontrast zwischen dem Vordergrund und dem Hintergrund. Je ungewohnter die Farben sind, desto besser finde ich.

![Image](https://github.com/user-attachments/assets/07ab2da1-74a9-4d35-8ca3-b4fa0fd94b1c)

**Comment by MarcHaunschild** [comment]
Also man stellt sich eigene Farben ein, damit etwas zugänglich wird was man vorher nicht nutzen konnte.
Dabei kommt es nicht auf Kontraste an.

Aber es dürfen auch keine Informationen verloren gehen.

Was oft passiert ist, dass bei eigenen Schriftgrößen Inhalte abgeschnitten werden oder dass Custome Radio-Buttons verwendet werden, die per Hintergrundfarben gestaltet sind und dann unsichtbar werden.

In der Beschreibung heißt es:

> Prüfen, ob alle wichtigen Bedienelemente und deren Zustände (z.B. "Hamburger"-Schaltfläche und Zustände von Formularelementen wie Auswahlschalter und Kontrollkästchen) bei geänderten Vorder- und Hintergrund-Farben noch sichtbar sind.

Was wäre denn der Sinn der Anforderung, wenn man nur prüft ob die Einstellungen übernommen werden, aber danach nichts mehr funktioniert?

Das wäre ja ein Feature ohne praktischen Nutzen. die Autoren der EN werden es aber doch nicht allen Entwicklern von Websites zur Pflicht gemacht, das sich das Design von Webseiten zum Spaß kaputt machen lässt. Hier hat man (wie bei allen anderen Anforderungen) einen Vorteil für Nutzer im Sinn gehabt.

**Comment by detlevhfischer** [comment]
@MarcHaunschild

Die Grenze ist schwierig zu ziehen. Durch Einstellungen kann alles Mögliche nicht mehr OK sein und wenn wir das berücksichtigen, wird es zwangeswiese viele subjektive Einschätzungen und unterscheidliche Bewertungen geben. Für den einen ist ein leicht abgeschnittener Text dann noch OK, die nächste nennt das schon FAIL. 

Klar ist der Fall z.B. da, wo durch *jede* Einstellung anderer Farben eine wichtige Infiormation wie de rAuswahlzustand der Custom-Checkboxen komplett verschwindet, oder Texte überhaupt nicht vergrößerbar sind. 
In anderen Fällen (z.B. Hamburger Icon ohne eigenen Hintergund bzw. ohne Nutzung von CSS current-color) wird es schwieriger.

**Comment by MarcHaunschild** [comment]
Ja, wenn bei eigenen Farben (egal welche) Radio-Buttons komplett verschwinden, ist es klar.

Und ja bei anderen ist es schwierig zu bewerten und kann zu Diskussionen zwischen Prüfer und QS führen.
😉

Es ist halt so ein winziger Aufwand einem Hamburger-Icon eine Farbe mitzugeben oder wenn es ein SVG ist, currentColor mitzugeben, dass es schade ist wenn so die Hauptnavigation(! - nicht irgendein Firlefanz) unerreichbar wird.

Und ja, ich weiß auch keine Formulierung für so etwas.

Insgesamt meine ich aber doch, dass diese Anforderung existiert, damit sich Menschen eine Website anpassen und dann besser nutzen können. Nicht schlechter.

Irgendwie muss sich die Absicht der Autoren in einem Prüfergebnis widerspiegeln.

Wie immer: meine 2 Cent und ich werde unabhängig von meiner Sichtweise so bewerten, wie wir es hier festlegen, damit wir vergleichbare Ergebnisse produzieren.

Abschließend: **ich bin keinesfalls dafür, den Prüfschritt noch mal grundlegend zu überarbeiten vor der neuen EN-Version**.

Hoffen wir auf mehr Klarheit in 2026 und passen dann entsprechend an.

**Comment by detlevhfischer** [comment]
@MarcHaunschild OK, warten wir ab. Es war ja noch unklar, ob es ein von 11.7 abgeleitete neue Anforderung 9.7 (für den Bereich Web) geben wird oder nicht. Nach dem Text, der jetzt anscheinend in den Entwurf der Überarbeitung übernommen wurde, steht als normative Forderung nicht mehr als 
> shall follow the values of the preferences that users have set

...und **nichts** darüber, was für ein Zustand dadurch auf der Seite dann produziert wird. Wenn wir z.B. wegen fehlender Checkbox-Zustände auf FAIL gehen, gehen wir vielleicht sogar schon über die normative Forderung hinaus. Auch wenn es aus Nutzenden-Sicht komplett nachvollziehbar ist.

**Comment by sweckenmann** [comment]
@detlevhfischer In dem Etsi Repo steht:

> Where ICT is or includes non-web software that provides a user interface, and is not designed to be isolated from its platform,
that user interface shall follow the values of the preferences that users have set for the platform documented accessibility features.

und dann:

> It was agreed in the 15th January meeting of STF614 to insert this in latest draft of the EN.

d.h. die Anforderung wäre für Web nicht mehr vorhanden (außer es wird eine neue Anforderung unter 9.7 eingeführt)?

Ist aber noch nicht im "stable draft" V3.1.1 (vermutlich noch nicht aktualisiert):

> Where software is not designed to be isolated from its platform ...

**Comment by MarcHaunschild** [comment]
Ich frage mich eh, wie sinnvoll diese Anforderung ist angesichts der Tatsache, dass inzwischen alle Browser eine sehr aufgeräumte und an individuelle Bedürfnisse anpassbare Reader-Darstellung haben.

Für Nutzer scheint es hilfreicher zu sein, wenn Präferenzen im Betriebssystem statt im Browser berücksichtigt würden (dark mode) oder eben der Reader-Modus funktionieren soll.

Beides ist aktuell nicht gefordert.

**Comment by johannesFischer84** [comment]
> Das issue ist ja schon älter (+4 Jahre) - im Prüfschrittt gibt es inzwischen eine Anleitung für die Einstellungen in Firefox. Bist Du damit soweit einverstanden? Dann könntest Du dieses Issue schließen,

Ja, das Issue hat sich erledigt. Ich schließe es.

Zu der zusätzlichen Frage von heute: Eine bestimmte Farbkombination finde ich nicht entscheidend. Ich denke, wichtig ist, dass Farben übernommen werden und dass bestimmte Bedienelemente nicht verschwinden, weil sie transparent sind. Die Kontraste der ausgewählten Farben sollten aber keine Rolle spielen.
